src/main/java/com/example/springboottask/controller/UserController.java

```diff
@@ -15,42 +19,54 @@
 import org.springframework.web.bind.annotation.RequestBody;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping("/api/users")
 @RequiredArgsConstructor
 public class UserController {
+
     private final UserService userService;
 
     private final OrderService orderService;
 
+    private final UserDtoConverter userDtoConverter;
+
+    private final OrderDtoConverter orderDtoConverter;
+
     @GetMapping
-    public List<User> getUsers() {
-        return userService.getUsers();
+    public List<UserDto> getUsers() {
+        List<User> users = userService.getUsers();
+        return users.stream()
+                .map(user -> userDtoConverter.convertToDto(user))
+                .collect(Collectors.toList());
     }
 
     @GetMapping("/{userId}")
-    public User getUser(@PathVariable Long userId) {
+    public UserDto getUser(@PathVariable Long userId) {
 
         User user = userService.getUser(userId);
+        UserDto userDto = userDtoConverter.convertToDto(user);
 
-        if (user == null) {
+        if (userDto == null) {
             throw new EntityResultNotFoundException("User id not found - " + userId);
         }
 
-        return user;
+        return userDto;
     }
 
     @PostMapping
-    public User addUser(@RequestBody User user) {
-        userService.saveUser(user);
+    public UserDto addUser(@RequestBody UserDto userDto) {
+        userService.saveUser(userDtoConverter.convertToModel(userDto));
 
-        return user;
+        return userDto;
     }
 
     @PutMapping
```

Comment:
userId should be provided as a path variable and then used to retrieve user from DB